src/libsyntax/ast.rs

```diff
@@ -456,7 +472,7 @@ pub enum MetaItemKind {
     /// List meta item.
     ///
     /// E.g. `derive(..)` as in `#[derive(..)]`
-    List(InternedString, Vec<P<MetaItem>>),
+    List(InternedString, Vec<MetaListItem>),
```

Comment:
I am really dubious about removing this pointer. It means we're going to clone a goodly number of literals, which might cause perf regressions in, eg, serde.